src/main/java/net/starlark/java/eval/StringModule.java

```diff
@@ -285,11 +298,25 @@ public String rstrip(String self, Object charsOrNone) {
             },
             doc = "The characters to remove, or all whitespace if None.",
             defaultValue = "None")
-      })
-  public String strip(String self, Object charsOrNone) {
-    CharMatcher matcher =
-        charsOrNone != Starlark.NONE ? CharMatcher.anyOf((String) charsOrNone) : LATIN1_WHITESPACE;
-    return stringStrip(self, matcher);
+      },
+      useStarlarkThread = true)
+  public String strip(String self, Object charsOrNone, StarlarkThread starlarkThread) {
+    return stripSemantics(self, charsOrNone, starlarkThread.getSemantics());
+  }
+
+  public String stripSemantics(
+      String self, Object charsOrNone, StarlarkSemantics starlarkSemantics) {
+    return stringStrip(self, matcher(charsOrNone, starlarkSemantics));
+  }
+
+  private static CharMatcher matcher(Object charsOrNone, StarlarkSemantics starlarkSemantics) {
+    return charsOrNone != Starlark.NONE
+        // TODO: This doesn't work correctly for non-ASCII characters in charsOrNone if using UTF-8
```

Comment:
Nit: It's a little ambiguous what the comment is saying, how about:

> When using the latin-1 hack, each utf-8 code unit is stored as a distinct string element. To avoid matching an element that doesn't correspond to a whole code point, we exclude anything that's not in the ASCII range.